src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.EasyRequest.cs

```diff
@@ -199,11 +199,20 @@ public void FailRequest(Exception error)
 
             private void SetUrl()
             {
-                EventSourceTrace("Url: {0}", _requestMessage.RequestUri);
-                SetCurlOption(CURLoption.CURLOPT_URL, _requestMessage.RequestUri.AbsoluteUri);
+                string url = GetAbsoluteUri(_requestMessage.RequestUri);
+                EventSourceTrace("Url: {0}", url);
+                SetCurlOption(CURLoption.CURLOPT_URL, url);
                 SetCurlOption(CURLoption.CURLOPT_PROTOCOLS, (long)(CurlProtocols.CURLPROTO_HTTP | CurlProtocols.CURLPROTO_HTTPS));
             }
 
+            private static string GetAbsoluteUri(Uri uri)
+            {
+                // uri.AbsoluteUri/ToString() do not include IPv6 scope IDs, but when provided they
+                // should be used or else libcurl may be unable to correctly connect.  SerializationInfoString
+                // ensures these details are included.
+                return uri.GetComponents(UriComponents.SerializationInfoString, UriFormat.UriEscaped);
```

Comment:
So, while this mechanism appears to work to include IPV6 link-local scope ids, I worry that it might not do the right thing if internationalized (i.e. Unicode) type of Uri's are used. While I can't prove it right now, the .AbsoluteUri method will potentially differ from the string you're getting using:

```c#
uri.GetComponents(UriComponents.SerializationInfoString, UriFormat.UriEscaped);
```


In fact, we have very little testing at all for this even for Windows implementation in CoreFx. For example, the Windows stack tends to want to have only Punnycode encoded Uri if the input strings contain Unicode code points. Most HTTP stacks don't like to receive non-ASCII values especially for DNS labels in the hostname portion of a uri.

Is it possible, to mitigate risk, to only use this logic iff. there is a scope ID present in the URI?  If not, use the original .AbsoluteUri method?

Not directly related to this PR, but in any case, we don't have a lot of testing in CoreFx with HTTP endpoints using internationalized URIs and we need to find a good way to improve that.